code/game/turfs/simulated/floor.dm

```diff
@@ -42,6 +42,12 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
 	if(floor_tile)
 		builtin_tile = new floor_tile
 
+/turf/simulated/floor/Destroy()
+	if(builtin_tile)

```

Comment:
if qdel stops checking for nulls that would be a bug, and ideally we would notice it there instead of having it silently stopped here.